### ISSUE TITLE
Feature Request: registry key for disabling saving passwords

### ISSUE THREAD

**Issue by bluikko** [mRemoteNG/mRemoteNG#738]
mRemoteNG could have an option in registry for disabling saving passwords. This way in enterprise setting a GPO could be deployed to disable the users from saving passwords.

I can volunteer to do the GPO-side work if this would be possible.

**Comment by srsirius** [comment]
Hi. 

On my remote server, I use a VNC 'reflector/repeater' which operates in 'Mode I' and receives my incoming VNC connection, then forwards it to a particular client's VNC server. 

The way this works is.... say my remote server is W.X.Y.Z, and the target client I want to admin is A.B.C.D With UltraVNC viewer, I specify A.B.C.D in the 'VNC Server' field (synonymous with 'Hostname/IP' in mRemoteNG), and then there is a field called 'Proxy/Repeater' in which I specify the reflector/repeater server W.X.Y.Z

**Comment by bluikko** [comment]
> Hi.
> 
> On my remote server, I use a VNC 'reflector/repeater' which operates in 'Mode I' and receives my incoming VNC connection, then forwards it to a particular client's VNC server.

Did you reply to wrong issue?

This issue has nothing to do with VNC. This issue is to request a way to disable saving passwords - for enforcing policy that users cannot save passwords.

**Comment by c0154936** [comment]
Lots of IT Security dept do not allow the use of RDP Manager tools because of the password saving ability. I also miss this tool but if we can disable the password saving ability, we would have it re-installed for all of us IT Admins. Even password protecting the config file does not satisfy our security dept.

**Comment by daughk** [comment]
Hoping that this feature request can get some traction.  Encrypting the connection file is a great start but we need the ability to disable password storage altogether.

**Comment by O815163** [comment]
> Hoping that this feature request can get some traction. Encrypting the connection file is a great start but we need the ability to disable password storage altogether.

same here

**Comment by SimplifyAndAddCoffee** [comment]
+1 to this. We need the ability to disable credential storage to comply with security policies.

**Comment by Kvarkas** [comment]
some options are needed for company only not home users, seems we need split mRemoteNG into 2 target builds - perhaps Core version and Business related...

but disable password saving should be implemented with profiles, a lot of abilities to control how connection may happened incliuding user separation, planed in build 1.8...

**Comment by bluikko** [comment]
> should be implemented with profiles

The reason why this issue specifically mentions the registry is that it will be easy to change this setting from outside of mRemoteNG, For example with a `.reg` file or a GPO.

**Comment by Kvarkas** [comment]
yes, curently we are using settings files and will have a look if I could add setting to enable registry to save them instead of local copy, but not in coming nb yet

**Comment by simonai1254** [comment]
> The reason why this issue specifically mentions the registry is that it will be easy to change this setting from outside of mRemoteNG, For example with a `.reg` file or a GPO.

I get that it's easy to change a registry key, but on the other hand protecting the registry key is hard. And adding the registry key check into mRemoteNG that currently uses a file based configuration script needs some work. 
Currently I would think more of a "locked configuration" file like you can implement for Firefox with the AutoConfig. The only thing with that solution is that it's a little harder to deploy the configuration in a precautionary manner than with a registry key... 

I guess the want-it-all solution would be to offer a dedicated GPO-package, but this might be thinking a little too far ahead 😊

**Comment by bluikko** [comment]
> but on the other hand protecting the registry key is hard.

Why would that key need to be protected? The registry key could be in a part of the registry where user has no write permissions. Or even with no controls at all, GPP is re-applied every 90 (by default) minutes.

**Comment by robertpopa22** [comment]
Triage refresh (P4 version-label debt): this issue is tagged with legacy version labels. Please retest on the latest build (fork v1.79.x or upstream dev), then share reproducible steps and expected vs actual behavior. If still valid, keep open and relabel to current triage scope (for example Need 2 check). If fixed, close with resolution.

Current legacy version labels: 1.78.*

**Comment by robertpopa22** [comment]
**Fix available for testing**

**Commit:** [`9a057eb0`](https://github.com/robertpopa22/mRemoteNG/commit/9a057eb0b0bb890eaa1754bf83212f3b12f2f5c5)
**Branch:** `main`
**What changed:** Implement a check for a 'DisablePasswordSaving' registry key

**Download latest beta:** [v1.81.0-beta.2](https://github.com/robertpopa22/mRemoteNG/releases/tag/v1.81.0-beta.2)

Please test and report if this resolves your issue.

---
_Automated by mRemoteNG Issue Intelligence System_

**Comment by bluikko** [comment]
Nice. Reviewing the code this looks gtg so far. The HKLM hive is used which is the right way. Will test.

Would you like to host `.admx`/`.adml` Policy Definition files somewhere? These could be loaded to the AD DS Central Store and then the normal GPO editing GUI tool (GPMC) workflow could be used to change these settings -- instead of adding a registry key GPP.  
I'll probably look at making them at some point soon and will host in my own repo at least.
